src/plugins/janus_streaming.c

```diff
@@ -9316,6 +9316,25 @@ static void *janus_streaming_relay_thread(void *data) {
 		}
 	}
 
+	/* Close the ports we bound to */
+	temp = source->media;
+	while(temp) {
+		janus_streaming_rtp_source_stream *stream = (janus_streaming_rtp_source_stream *)temp->data;
+		if(stream->fd[0] > -1)
+			close(stream->fd[0]);
+		stream->fd[0] = -1;
+		if(stream->fd[1] > -1)
+			close(stream->fd[1]);
+		stream->fd[1] = -1;
+		if(stream->fd[2] > -1)
+			close(stream->fd[2]);
+		stream->fd[2] = -1;
```

Comment:
This looks a bit redundant, why not use a simple loop?
```c
	for(int i=0; i<3; i++) {
  		if(stream->fd[i] >= 0) {
			close(stream->fd[i]);
  			stream->fd[i] = -1;
		}
	}
```

Reply:
Since i was small enough and known, it felt easier to do that. Besides, I just copied what we had in the _free callback already.